ruler/configs.go

```diff
@@ -0,0 +1,111 @@
+package ruler
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+
+	"github.com/prometheus/common/log"
+	"github.com/prometheus/prometheus/promql"
+	"github.com/prometheus/prometheus/rules"
+)
+
+// TODO: Extract configs client logic into go client library (ala users)
+
+type configID int
+
+type cortexConfig struct {
+	RulesFiles map[string]string `json:"rules_files"`
+}
+
+type cortexConfigView struct {
```

Comment:
To understand this type and the other code in this file, it would be helpful to document the assumptions which I hopefully correctly derived after reading it and related code: that subsequent config revisions for the same org (apparently) get new config IDs, and that these IDs are monotonically increasing, and that this fact is used to fetch only the latest config updates from the config API.